pybedlite/overlap_detector.py

```diff
@@ -322,18 +322,38 @@ def overlaps_any(self, interval: Span) -> bool:
             True if and only if the given interval overlaps with any interval in this
             detector.
         """
-        tree = self._refname_to_tree.get(interval.refname)
+        tree = self._refname_to_tree.get(interval.refname, None)
         if tree is None:
             return False
         else:
             if not self._refname_to_indexed[interval.refname]:
                 tree.index()
-            try:
-                next(iter(tree.overlap(interval.refname, interval.start, interval.end)))
-            except StopIteration:
-                return False
-            else:
-                return True
+                self._refname_to_indexed[interval.refname] = True
+            # IntervalSet uses closed intervals whereas we are using half-open intervals, so add 1
+            # to start
+            return tree.any_overlaps(interval.start + 1, interval.end)
+
+    def iter_overlaps(self, interval: Span) -> Iterator[SpanType]:
```

Comment:
I think this method should probably be private, otherwise there's redundancy with `get_overlaps()`.

If we'd like to permit the user to access unsorted overlaps, or receive an iterator instead of a list, I think we can do those within `get_overlaps()`.

Reply:
I responded to this above so the discussion could be in one thread.